homework_weekend7/src/task1/LinkedListBoolean.java

```diff
@@ -0,0 +1,198 @@
+package task1;
+
+import java.util.NoSuchElementException;
+
+/**
+ * Represents implementation of LinkedList for boolean values. Implements common
+ * methods related to the LinkedList.
+ * 
+ * @author ajla
+ *
+ */
+
+public class LinkedListBoolean {
+	private Node head;
+
+	/**
+	 * Adds element at the end of the list.
+	 * 
+	 * @param value
+	 * @return
+	 */
+	public void add(boolean value) {
+		if (head == null) {
+			head = new Node(value);
+			return;
+		}
+
+		Node tmp = head;
+
+		while (tmp.getNextNode() != null) {
+			tmp = tmp.getNextNode();
+		}
+		tmp.setNextNode(new Node(value));
+	}
+
+	/**
+	 * Removes first element with given value.
+	 * 
+	 * @param elem
+	 * @throws NoSuchElementException
+	 */
+	public void remove(boolean elem) throws NoSuchElementException {
```

Comment:
Better signature for this method would be to return boolean value. Than in case of success, true is returned, else false. Also this would allow for not resorting to using exceptions for handling cases where there is nothing to remove.